rm/rm-server/src/main/java/org/ow2/proactive/resourcemanager/core/RMCore.java

```diff
@@ -1356,37 +1369,129 @@ public BooleanWrapper deployNodeSource(String nodeSourceName) {
 
         RMCore.clients.put(nsId, nsService);
         RMCore.clients.put(policyId, policyService);
+    }
 
-        updateNodeSourceInfo(nodeSourceToDeploy);
+    private NodeSourcePolicy createNodeSourcePolicyActivity(NodeSourceDescriptor nodeSourceDescriptor,
+            NodeSource nodeSourceToDeploy) {
 
-        this.monitoring.nodeSourceEvent(new RMNodeSourceEvent(RMEventType.NODESOURCE_CREATED,
-                                                              provider.getName(),
-                                                              nodeSourceToDeploy.getName(),
-                                                              nodeSourceToDeploy.getDescription(),
-                                                              nodeSourceToDeploy.getAdministrator().getName(),
-                                                              nodeSourceToDeploy.getDescriptor()
-                                                                                .getStatus()
-                                                                                .toString()));
+        NodeSourcePolicy nodeSourcePolicyStub = NodeSourcePolicyFactory.activate(nodeSourceToDeploy.getPolicy(),
+                                                                                 nodeSourceDescriptor.getPolicyParameters());
 
-        logger.info("The node source " + nodeSourceName + " has been successfully deployed by " + provider);
+        nodeSourceToDeploy.setActivePolicy(nodeSourcePolicyStub);
 
-        return new BooleanWrapper(true);
+        return nodeSourcePolicyStub;
+    }
+
+    private NodeSource createNodeSourceActivity(String nodeSourceName, NodeSource nodeSourceToDeploy) {
+
+        try {
+            nodeSourceToDeploy = PAActiveObject.turnActive(nodeSourceToDeploy, this.nodeRM);
+        } catch (Exception e) {
+            logger.error(e.getMessage(), e);
+            throw new RuntimeException("Cannot create node source activity " + nodeSourceName, e);
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Define and throw a dedicated exception instead of using a generic one. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS00112)